### PR TITLE
Pipeline updates

### PR DESCRIPTION
**Pull request by jvntf**
- [x] fix a typo
- [x] make datetime regex an input to s3-discovery task
- [x] enhance regex parsing input specification
- [x] add permission definition to the cdk workflow
- [x] change stac gen + db insert to happen in 2 steps via chaining / passing stac item json to insertion lambda

### CODE REVIEW

**Comment by jvntf** [comment]
@abarciauskas-bgse this can be reviewed in.
The current implementation separates the stac item generation and the db insertion into 2 lambdas which are chained in the state machine. 

I'd like to merge this and then implement an architecture advised by @leothomas and @anayeaye:
`cog-pipeline` stac will generate stac records and write to s3 as a temporary store, and queue up a message with SQS. we can then add a lambda to `delta-backend` which polls for messages, writes new stac records to the database, and then deletes the temp files from s3.

File: db-write/handler.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+
+def handler(event, context):
+    """
+    Simple Lambda that should belong to the VPC of the STAC_DB_HOST
+    Receive a STAC item in JSON format, connect to database and insert it
+    """
+    stac_json = event["stac_item"]
+    # pypgstac requires inserting from a file
+    with open("/tmp/temp.json", "w+") as f:
```
Comment (abarciauskas-bgse) [inline db-write/handler.py:24]: nit - name this file something more descriptive like `/tmp/stac_item.json`

Comment (abarciauskas-bgse) [reply db-write/handler.py:24]: Also, the filename should be a variable:

```
stac_item_filename = "/tmp/stac_item.json"
with open(stac_item_filename, "w+") as f:
...
pypgstac.load(
    table="items",
    file = stac_item_filename, ...)

os.remove(stac_item_filename)
```

File: s3-discovery/handler.py [hunk 2]
```diff
@@ -34,10 +34,7 @@ def handler(event, context):
...
                 "collection": event["prefix"][:-1],
                 "s3_filename": f's3://{event["bucket"]}/{f}',
-                "datetime_regex": {
-                    "regex": f"^(.*?)(_)([0-9][0-9][0-9][0-9])({event['file_type'})$",
-                    "target_group": 3,
-                },
+                "datetime_regex": event["datetime_regex"],
             }
         )
     print(files_objs)
```
Comment (abarciauskas-bgse) [inline s3-discovery/handler.py:40]: should we remove this debug statement?

File: cdk/cdk/cdk_stack.py [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+from aws_cdk import core, aws_iam
 import aws_cdk.aws_stepfunctions as stepfunctions
 import aws_cdk.aws_events as events
 import aws_cdk.aws_events_targets as targets
 from aws_cdk import aws_lambda
 from aws_cdk import aws_stepfunctions_tasks as tasks
+from aws_cdk import aws_ec2 as ec2
 import os
 
+VPC_ID = "vpc-0b0926e1be04a588d"
```
Comment (abarciauskas-bgse) [inline cdk/cdk/cdk_stack.py:10]: Is there a reason why this is hard-coded or can we use an environment variable?

**Comment by abarciauskas-bgse** [comment]
This looks good @jvntf just made a few minor comments.

I think the proposed infrastructure sounds good but I think there may be a few options we consider as a group Monday. If we're PUT'ting stac json to S3, this could be used as an event trigger for Lambda, however there may be a good reason to use the additional SQS queue. In case we want to create the SQS approach (and you may already know about this) this might be helpful documentation about how to do so: https://docs.aws.amazon.com/AmazonS3/latest/userguide/ways-to-add-notification-config-to-bucket.html#step2-enable-notification

**Comment by abarciauskas-bgse** [comment]
To make a few points more clear:
* If we're writing a STAC JSON file to S3, this `S3:PutObject` event could just trigger the db-write lambda (you already know this I'm stating just for clarity)
* We may decide we want an SQS queue to manage any failed lambda executions of db-write with additional retrying and placing failures after XX tries in a dead-letter queue. But I am curious if there are other reasons to use a queue which include writing to the same queue from multiple sources. However in this case I'm curious about how we handle priorities and failures, for example we don't want one sources to dump a bunch of messages on the queue and block a smaller job.

**Review by abarciauskas-bgse — APPROVED** [review]

**Pull request merged**